plugins/reindex/src/main/java/org/elasticsearch/plugin/reindex/RestUpdateByQueryAction.java

```diff
@@ -0,0 +1,94 @@
+package org.elasticsearch.plugin.reindex;
+
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.indices.query.IndicesQueriesRegistry;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestChannel;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.search.RestSearchAction;
+import org.elasticsearch.rest.action.support.RestActions;
+import org.elasticsearch.script.Script;
+
+import java.util.Map;
+
+import static org.elasticsearch.plugin.reindex.RestReindexAction.parseCommon;
+import static org.elasticsearch.plugin.reindex.UpdateByQueryAction.INSTANCE;
+import static org.elasticsearch.rest.RestRequest.Method.POST;
+
+public class RestUpdateByQueryAction extends BaseRestHandler {
+    private IndicesQueriesRegistry indicesQueriesRegistry;
+
+    @Inject
+    public RestUpdateByQueryAction(Settings settings, RestController controller, Client client,
+            IndicesQueriesRegistry indicesQueriesRegistry) {
+        super(settings, controller, client);
+        this.indicesQueriesRegistry = indicesQueriesRegistry;
+        controller.registerHandler(POST, "/{index}/_update_by_query", this);
```

Comment:
Not related to your PR, but this [lets the `this` pointer escape](http://www.javaspecialists.eu/archive/Issue192.html). I think it would be good to have a separate method that is run after the constructor where we can do such things safely.